### DISCUSSION TITLE
UART seems to be dead in MicroPython but works with C code

### DISCUSSION THREAD

**Discussion by vpmdt** [micropython/micropython#11848] [RP2040 / Pico]
Hi, 
I was able to read/write over uart until today in MicroPython. No idea what happend. If I upload .uf2 file with hello world in C, I can read it in Putty, but I am getting no hello world with the MicroPython. This is for proving that, the usb to serial debug cable is ok. I am using GPI0 pins 0, 1 and GND. I am using Thonny 4.1.1 for MicroPython scripts. Any ideas why it stopped working?? Maybe Thonny for some reason is not configuring the uart pins?

**Comment by shariltumin** [comment]
You can try this. In Thonny Shell i.e. REPL check that uart is working.
This is what I got:

```
MicroPython v1.20.0-242-g6a9db521e on 2023-06-30; Raspberry Pi Pico with RP2040
Type "help()" for more information.
>>> from machine import UART, Pin
>>> uart = UART(0, baudrate=9600, tx=Pin(0), rx=Pin(1))
>>> uart.write('hello world')
11
>>> uart.write('hello world')
11
>>> w = uart.read()
>>> print(w)
b'hello worldhello world'
>>>
```

You will need to connect pin 0 and pin 1 (i.e. connect TX to RX) of your board.
